src-extras/Database/LSMTree/Extras/MergingRunData.hs

```diff
@@ -0,0 +1,217 @@
+-- | Utilities for generating 'MergingRun's. Tests and benchmarks should
+-- preferably use these utilities instead of (re-)defining their own.
+module Database.LSMTree.Extras.MergingRunData (
+    -- * Flush merging runs
+    withMergingRun
+  , unsafeFlushMergingRun
+    -- * MergingRunData
+  , MergingRunData (..)
+  , mergingRunDataInvariant
+  , mapMergingRunData
+  , SerialisedMergingRunData
+  , serialiseMergingRunData
+    -- * QuickCheck
+  , labelMergingRunData
+  , genMergingRunData
+  , shrinkMergingRunData
+  ) where
+
+import           Control.Exception (bracket)
+import           Control.Monad (forM)
+import           Control.RefCount
+import qualified Data.Vector as V
+import           Database.LSMTree.Extras (showPowersOf)
+import           Database.LSMTree.Extras.Generators ()
+import           Database.LSMTree.Extras.RunData
+import           Database.LSMTree.Internal.Index (IndexType)
+import           Database.LSMTree.Internal.Lookup (ResolveSerialisedValue)
+import           Database.LSMTree.Internal.MergingRun (MergingRun)
+import qualified Database.LSMTree.Internal.MergingRun as MR
+import           Database.LSMTree.Internal.Paths
+import           Database.LSMTree.Internal.Run (RunDataCaching (..))
+import qualified Database.LSMTree.Internal.Run as Run
+import           Database.LSMTree.Internal.RunAcc (RunBloomFilterAlloc (..))
+import           Database.LSMTree.Internal.RunNumber
+import           Database.LSMTree.Internal.Serialise
+import           Database.LSMTree.Internal.UniqCounter
+import qualified System.FS.API as FS
+import           System.FS.API (HasFS)
+import           System.FS.BlockIO.API (HasBlockIO)
+import           Test.QuickCheck as QC
+
+{-------------------------------------------------------------------------------
+  Flush runs
+-------------------------------------------------------------------------------}
+
+-- | Create a temporary 'Run' using 'unsafeFlushMergingRun'.
+withMergingRun ::
+     MR.IsMergeType t
+  => HasFS IO h
+  -> HasBlockIO IO h
+  -> ResolveSerialisedValue
+  -> IndexType
+  -> FS.FsPath
+  -> UniqCounter IO
+  -> SerialisedMergingRunData t
+  -> (Ref (MergingRun t IO h) -> IO a)
+  -> IO a
+withMergingRun hfs hbio resolve indexType path counter mrd = do
+    bracket
+      (unsafeFlushMergingRun hfs hbio resolve indexType path counter mrd)
+      releaseRef
+
+-- | Flush serialised run data to disk as if it were a write buffer.
+--
+-- This might leak resources if not run with asynchronous exceptions masked.
+-- Use helper functions like 'withMergingRun' or 'withMergingRuns' instead.
+--
+-- Use of this function should be paired with a 'releaseRef'.
+unsafeFlushMergingRun ::
```

Comment:
"Flush" in my mind means flushing of the write buffer to disk. Maybe we should just call it `unsafeAsMergingRun`?

Reply:
or `unsafeCreateMergingRun`?

Reply:
I went with that.